From a74bceea7fee307ec38f21901f6803ae74b0b44c Mon Sep 17 00:00:00 2001
From: Huynh Thanh Hung <hung.huynh.wz@rvc.renesas.com>
Date: Tue, 12 Mar 2019 16:09:39 +0700
Subject: [PATCH 2/2] hmi: demos: encoded preview: change order of element due
 to vspfilter can't handle 2 format

Since in 1 pipeline, vspfilter can't handle 2 format, so using 1 format
for videosink and videoencode

Signed-off-by: Huynh Thanh Hung <hung.huynh.wz@rvc.renesas.com>
---
 RZ_GST-demo_encoded-review.sh | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/RZ_GST-demo_encoded-review.sh b/RZ_GST-demo_encoded-review.sh
index 0837cf1..cabc4fa 100755
--- a/RZ_GST-demo_encoded-review.sh
+++ b/RZ_GST-demo_encoded-review.sh
@@ -23,13 +23,14 @@ function record()
 	gst-launch-1.0 -e \
 	v4l2src num-buffers=$((TIME*30)) device=$CAM_DEV ! \
 	video/x-raw,format=UYVY,width=720,height=480 ! \
-	tee name=t t. ! \
 	vspfilter ! \
 	video/x-raw,format=NV12,width=640,height=480 ! \
+	queue ! tee name=t t. ! \
+	queue ! \
 	omxh264enc target-bitrate=10485760 ! \
 	h264parse ! video/x-h264,stream-format=avc,aligment=au ! qtmux ! queue ! \
 	filesink location=$1 t. ! \
-	vspfilter ! video/x-raw,format=BGRA,width=800,height=480 ! \
+	queue ! \
 	waylandsink qos=false
 }
 
-- 
2.7.4

